[doctest] run with "--help" for options
== TEST CASE ==================================================================
assertion_macros.cpp(0)
an empty test that will fail because of an exception

TEST CASE FAILED!
threw exception:
  0

== TEST CASE ==================================================================
assertion_macros.cpp(0)
an empty test that will fail because of a std::exception

TEST CASE FAILED!
threw exception:
  whops!

== TEST CASE ==================================================================
assertion_macros.cpp(0)
normal macros

assertion_macros.cpp(0) ERROR!
  CHECK( throw_if(true, std::runtime_error("whops!")) == 42 )
threw exception:
  whops!

assertion_macros.cpp(0) ERROR!
  CHECK( doctest::Approx(0.502) == 0.501 )
with expansion:
  CHECK( Approx( 0.502 ) == 0.501 )

TEST CASE FAILED!
threw exception:
  0

== TEST CASE ==================================================================
assertion_macros.cpp(0)
exceptions-related macros

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS( throw_if(false, 0) )
didn't throw at all

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS_AS( throw_if(false, 0), int )
didn't throw at all

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS_AS( throw_if(true, 0), char )
threw a different exception:
  0

assertion_macros.cpp(0) ERROR!
  CHECK_NOTHROW( throw_if(true, 0) )
threw exception:
  0

== TEST CASE ==================================================================
assertion_macros.cpp(0)
exceptions-related macros for std::exception

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS( throw_if(false, 0) )
didn't throw at all

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS_AS( throw_if(false, std::runtime_error("whops!")), std::exception& )
didn't throw at all

assertion_macros.cpp(0) ERROR!
  CHECK_THROWS_AS( throw_if(true, std::runtime_error("whops!")), int )
threw a different exception:
  whops!

assertion_macros.cpp(0) FATAL ERROR!
  REQUIRE_NOTHROW( throw_if(true, std::runtime_error("whops!")) )
threw exception:
  whops!

===============================================================================
[doctest] test cases:    6 |    1 passed |    5 failed | 
[doctest] assertions:   17 |    7 passed |   10 failed |
Program code.
